fastparse/jvm/src/test/scala/fastparse/MidiTests.scala

```diff
@@ -0,0 +1,112 @@
+package fastparse
+
+import java.nio.file.{Files, Paths}
+import javax.sound.midi.MidiSystem
+
+import utest._
+
+
+object MidiTests extends TestSuite{
+  def readResourceBytes(file: String) = {
+    Files.readAllBytes(Paths.get(getClass.getResource(file).toURI.getPath))
+  }
+
+  def hexBytes(bytes: Array[Byte]) = {
+    println(" \t" + 0.until(16).map(x => " " * (if (x >= 10) 0 else 1) + x).mkString(" "))
+    println()
+
+    println(
+      bytes
+        .take(512)
+        .grouped(16)
+        .zipWithIndex
+        .map{case (v, i) => (i * 16) + "\t" + ElemTypeFormatter.ByteFormatter.prettyPrint(v)}
+        .mkString("\n")
+    )
+  }
+
+  def variousParses(bytes: Array[Byte]) = Seq(
+    MidiParse.midiParser.parse(bytes).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(1)).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(4)).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(16)).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(64)).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(256)).get.value,
+    MidiParse.midiParser.parseIterator(bytes.grouped(1024)).get.value
```

Comment:
`for (size <- Seq(1, 4, 16, ...)) MidiParse.midiParser.parseIterator(bytes.grouped(size)).get.value`
Not like that? :)